src/util/system.cpp

```diff
@@ -1079,6 +1083,39 @@ void runCommand(const std::string& strCommand)
 }
 #endif
 
+#ifdef HAVE_BOOST_PROCESS
+UniValue RunCommandParseJSON(const std::string& str_command, const std::string& str_std_in)
+{
+    namespace bp = boost::process;
+
+    UniValue result_json;
+    bp::opstream stdin_stream;
+    bp::ipstream stdout_stream;
+
+    if (str_command.empty()) return UniValue::VNULL;
+
+    bp::child c(
+        str_command,
+        bp::std_out > stdout_stream,
+        bp::std_in < stdin_stream
```

Comment:
In commit `[util] add RunCommandParseJSON`

We should be piping stderr too. In HWI, this is used for some info to the user that we can discard here. But on failure, it can provide useful debugging information that we would want to see in the exception.